app/controllers/assets_controller.rb

```diff
@@ -265,12 +283,9 @@ def load_vars
     @asset = Asset.find_by_id(params[:id])
     return render_404 unless @asset
 
-    step_assoc = @asset.step
-    result_assoc = @asset.result
-    repository_cell_assoc = @asset.repository_cell
-    @assoc = step_assoc unless step_assoc.nil?
-    @assoc = result_assoc unless result_assoc.nil?
-    @assoc = repository_cell_assoc unless repository_cell_assoc.nil?
+    @assoc ||= @asset.step
+    @assoc ||= @asset.result
+    @assoc ||= @asset.repository_cell
 
```

Comment:
I was starring at this for some time, i'ts brilliant.

What about good old @assoc = @asset.step || @asset.result  || @asset.repository_cell

or even
assoc= [@asset.step, @asset.result, @asset.repository_cell].compact.first
(the latter is obviously not showing our intention clearly though)